config/cron-tasks.ts

```diff
@@ -0,0 +1,56 @@
+export default {
+  /**
+   * Delete notifications that are over a week past their due date and orphaned notifications daily at midnight
+   */
+  cleanupExpiredNotifications: {
+    task: async ({ strapi }) => {
+      console.log("Running cleanupExpiredNotifications task");
+      try {
+        // Calculate date 1 week ago
+        const oneWeekAgo = new Date();
+        oneWeekAgo.setDate(oneWeekAgo.getDate() - 7);
+
+        // Find all notifications with templates due over a week ago
+        const expiredNotifications = await strapi.db
```

Comment:
Probably it's not a case.
But if there are too many of `expiredNotifications` it might be a heavy request bellow (deleteMany). We might consider batch deleting (up to 10k of elements at once)